code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -636,3 +630,32 @@
 	desc = "Allows you to to turn a cyborg into a clown, honk."
 	icon_state = "cyborg_upgrade3"
 	new_module = /obj/item/robot_module/clown
+
+/obj/item/borg/upgrade/research
+	name = "research upgrade"
+	desc = "An efficent module that uses the cyborg processor to calculate complex math equations, uses extra power to run."
+	icon_state = "graphic_card"
+	var/mob/living/silicon/robot/cyborg
+
+/obj/item/borg/upgrade/research/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		var/obj/item/borg/upgrade/research/U = locate() in R
+		cyborg = R
+		if(U)
+			to_chat(user, "<span class='warning'>This unit is already equipped with a research module.</span>")
+			return FALSE

```

Comment:
Make it prevent insertion if the borg already has a research module

Reply:
what

Reply:
The way you set it up, if i read it correctly, still consumes the module even if the borg already has one.

Reply:
it drops it on the floor gonna test it again to be 100% sure

Reply:
tested it